beacon-chain/blockchain/process_block.go

```diff
@@ -518,11 +519,56 @@ func (s *Service) isDataAvailable(ctx context.Context, root [32]byte, signed int
 	if err != nil {
 		return errors.Wrap(err, "could not get KZG commitments")
 	}
+	existingBlobs := len(kzgCommitments)
+	if existingBlobs == 0 {
+		return nil
+	}
+
+	// Read first from db in case we have the blobs
+	s.blobNotifier.Lock()
+	var nc *blobNotifierChan
+	var ok bool
+	nc, ok = s.blobNotifier.chanForRoot[root]
 	sidecars, err := s.cfg.BeaconDB.BlobSidecarsByRoot(ctx, root)
-	if err != nil {
-		return errors.Wrap(err, "could not get blob sidecars")
```

Comment:
If err is not nil, shouldn't we also do something like return early?

Reply:
nvm, I think this is by design, perhaps it's safer to check the error for not found